python/mlir_structured/dialects/indexing.py

```diff
@@ -503,8 +500,32 @@ def _is_scalar(e: Any) -> bool:
   return isinstance(e, Scalar) or isinstance(e, (int, float, bool))
 
 
-def _canonicalize_tuple_index(idx, rank: int):
-  """Helper to remove Ellipsis and replace with implicit trailing slice(None)s.
+def _is_index_tensor(x):
+  """Returns True if x is a Tensor with index dtype, False otherwise."""
+  return (isinstance(x, Value) and Tensor.isinstance(x) and
```

Comment:
Is the first test needed? Doesn't `Tensor.isinstance` subsume that test?

Reply:
The first test is needed and `Tensor.isinstance` does not subsume - if you pass something that's not an `ir.Value` you will get a pybind11 exception from failing to cast [here](https://github.com/llvm/llvm-project/blob/2713781b0cdc1af647048ec97d40101664673dee/mlir/include/mlir/Bindings/Python/PybindAdaptors.h#L507).